src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb

```diff
@@ -96,10 +96,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
 
             If node.RelaxationLambdaOpt IsNot Nothing Then
-                returnValue = node.Update(VisitExpressionNode(node.RelaxationLambdaOpt),
+
+                If _inExpressionLambda AndAlso
+                    NoParameterRelaxation(DirectCast(node.Operand, BoundLambda).LambdaSymbol, node.RelaxationLambdaOpt.LambdaSymbol) Then
```

Comment:
At the top level there is always a lambda, but inside that lambda can be other things, including Anonymous Delegate conversions, I think. 

---
In reply to: [64994158](https://github.com/dotnet/roslyn/pull/11627#discussion_r64994158) [](ancestors = 64994158)